### DISCUSSION TITLE
Does this exporter also retrieve message IDs along with the text?

### DISCUSSION THREAD

**Discussion by SilSinn9801** [Tyrrrz/DiscordChatExporter#587] [Q&A] [answered]
Does DiscordChatExporter retrieve each message’s numeric ID (or alternatively its message link) & store it along with each message on the exported file? My interest isn’t much about retrieving all messages & images from a DM channel but more about retrieving all their message IDs, in case some of those messages are worth reporting to Discord Trust & Safety.

**Comment by 96-LB** [comment, marked as answer]
You can find the message ID in the raw HTML file right here:
![image](https://user-images.githubusercontent.com/21089343/118646884-c0b1c380-b7ae-11eb-8caa-f5d1c4299fe1.png)

The message ID is also readily available in the JSON export:
![image](https://user-images.githubusercontent.com/21089343/118647355-403f9280-b7af-11eb-95a3-6450de31b09d.png)
